src/Topologies/tensorproductgrids.jl

```diff
@@ -0,0 +1,32 @@
+function vertex_coordinates(
```

Comment:
why does this need to be in a separate file? would make sense to keep in `grids.jl`

Reply:
Okay, I thought it was more organized to have specific, concrete-typed methods in separate files, one for the regular case and one for the irregular one. I can keep them both in one file. 
Do you also prefer all the common methods of the abstract type together with the redefined methods for the concrete types all in one unique file?

Reply:
Done in [`82f89ed`](https://github.com/CliMA/ClimaCore.jl/pull/82/commits/82f89ed04ce33546240e0489473b174aad391484). I hope it is clearer now.